dids/src/main/kotlin/web5/sdk/dids/didcore/DIDDocument.kt

```diff
@@ -0,0 +1,295 @@
+package web5.sdk.dids.didcore
+
+import com.fasterxml.jackson.annotation.JsonProperty
+import java.security.SignatureException
+
+/**
+ * DIDDocument represents a set of data describing the DID subject including mechanisms such as:
+ * - cryptographic public keys - used to authenticate itself and prove association with the DID
+ * - services - means of communicating or interacting with the DID subject or
+ *   associated entities via one or more service endpoints.
+ * Examples include discovery services, agent services, social networking services, file storage services,
+ * and verifiable credential repository services.
+ * A DID Document can be retrieved by resolving a DID URI.
+ *
+ * @property id the DID URI for a particular DID subject, expressed using the id property in the DID document.
+ * @property context a URI that defines the schema version used in the document.
+ * @property alsoKnownAs AlsoKnownAs can contain multiple identifiers for different purposes,
+ *           or at different times for the same DID subject. The assertion that two or more DIDs
+ *           (or other types of URI) refer to the same DID subject can be made using the alsoKnownAs property.
+ * @property controller defines an entity that is authorized to make changes to a DID document.
+ * 	         The process of authorizing a DID controller is defined by the DID method.
+ * 	         It can be a string or a list of strings.
+ * @property verificationMethod a list of cryptographic public keys, which can be used to authenticate or authorize
+ * 	         interactions with the DID subject or associated parties.
+ * @property service expresses ways of communicating with the DID subject or associated entities.
+ * 	         A service can be any type of service the DID subject wants to advertise.
+ * @property assertionMethod used to specify how the DID subject is expected to express claims,
+ * 	         such as for the purposes of issuing a Verifiable Credential.
+ * @property authentication specifies how the DID subject is expected to be authenticated,
+ * 	         for purposes such as logging into a website or engaging in any sort of challenge-response protocol.
+ * @property keyAgreement specifies how an entity can generate encryption material to transmit confidential
+ * 	         information intended for the DID subject, such as for establishing a secure communication channel.
+ * @property capabilityDelegation specifies a mechanism used by the DID subject to delegate a
+ * 	         cryptographic capability to another party, such as delegating the authority to access a specific HTTP API.
+ * @property capabilityInvocation specifies a verification method used by the DID subject to invoke a
+ * 	         cryptographic capability, such as the authorization to update the DID Document.
+ */
+public class DIDDocument(
```

Comment:
In some placed we have `DidDocument` casing and other places (like here) we have `DIDDocument` casing. Any plans to make consistent?

Reply:
you probably saw `DidDocumentMetadata` - i changed that to `DIDDocumentMetadata` to make it consistent, ty!